plutus-ledger-api/test-onchain-evaluation/Main.hs

```diff
@@ -12,60 +13,71 @@ import PlutusLedgerApi.V1 qualified as V1
 import PlutusLedgerApi.V2 qualified as V2
 
 import Codec.Serialise qualified as CBOR
-import Data.Foldable (toList)
-import Data.Functor
+import Control.Concurrent.ParallelIO qualified as Concurrent
+import Control.Exception (evaluate)
+import Control.Monad.Extra (whenJust)
+import Data.List.NonEmpty (NonEmpty, nonEmpty, toList)
+import Data.Maybe (catMaybes)
+import PyF (fmt)
 import System.Directory.Extra (listFiles)
 import System.Environment (getEnv)
-import System.FilePath (isExtensionOf)
+import System.FilePath (isExtensionOf, takeBaseName)
 import Test.Tasty
 import Test.Tasty.HUnit
-import Test.Tasty.Providers
 
-data SingleTest
-    = SingleTest
-        LedgerPlutusVersion
-        (Maybe (EvaluationContext, [Integer]))
-        ScriptEvaluationEvent
+data TestFailure
```

Comment:
btw now that I think about it I think most of this stuff should live in the testlib with the definition of `ScriptEvaluationEvent`, since it's basically *the* thing you want to do with them.